### PR TITLE
Update Akka.Hosting extension methods to support DefaultAzureCredential

### LINKED ISSUE #221: Need to add default Azure credential support to Akka.Persistence.Azure.Hosting
https://github.com/petabridge/Akka.Persistence.Azure/pull/155 - need to add support for this for table storage and blob storage inside the new Akka.Persistence.Azure.Hosting library we're shipping.

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes #221
Add DefaultAzureCredential support to Akka.Hosting

### CODE REVIEW

File: src/Akka.Persistence.Azure.Hosting/AzurePersistenceExtensions.cs [hunk 1]
```diff
@@ -12,65 +15,318 @@ namespace Akka.Persistence.Azure.Hosting
     /// </summary>
     public static class AzurePersistenceExtensions
     {
-        public const string DefaultTableName = "AkkaPersistenceDefaultTable";
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Azure.Hosting/AzurePersistenceExtensions.cs:15]: We should leave these public - and are these the same default values that are also in HOCON?

File: src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs [hunk 2]
```diff
@@ -20,6 +21,7 @@ public class AzurePersistenceHostingSanityCheck
         public static async Task<IHost> StartHost(Action<AkkaConfigurationBuilder> testSetup)
         {
             var conn = Environment.GetEnvironmentVariable("AZURE_CONNECTION_STR") ?? "UseDevelopmentStorage=true";
+            await DbUtils.CleanupCloudTable(conn);
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs:24]: Can we add a separate test to this file that uses the `DefaultAzureCredential` ?

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Left some comments

File: src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs (hunk 2 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs:24]: We're hitting the same problem as Akka.Discovery.Azure here, `DefaultAzureCredential` needs HTTPS and bearer token to work, which Azurite does not support

File: src/Akka.Persistence.Azure.Hosting/AzurePersistenceExtensions.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Azure.Hosting/AzurePersistenceExtensions.cs:15]: Yes, those are the same default values in HOCON

File: src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/Akka.Persistence.Azure.Tests/Hosting/AzurePersistenceHostingSanityCheck.cs:24]: Ok, got it.

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**